### PR TITLE
Infra/refactor view - convert to function component and use modifiers and themeProps hooks

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
Convert View to a function component to reduce render hierarchy 

depends on https://github.com/wix/react-native-ui-lib/pull/2145

## Changelog
Convert View to a function component to reduce render hierarchy

### CODE REVIEW

File: src/components/view/index.tsx [hunk 1]
```diff
@@ -46,106 +40,90 @@ export interface ViewProps extends Omit<RNViewProps, 'style'>, ContainerModifier
...
-  alignments: true,
-  flex: true,
-  position: true
-};
+  return (
+    <ViewContainer
+      accessibilityElementsHidden={inaccessible}
+      importantForAccessibility={inaccessible ? 'no-hide-descendants' : undefined}
+      {...others}
+      style={[
```
Comment (M-i-k-e-l) [inline src/components/view/index.tsx:112]: We should probably memoize the `style`

File: src/components/view/index.tsx [hunk 2]
```diff
@@ -46,106 +40,90 @@ export interface ViewProps extends Omit<RNViewProps, 'style'>, ContainerModifier
...
-          alignments,
-          style
-        ]}
-        ref={forwardedRef}
-      >
-        {this.props.children}
-      </Element>
-    );
+    return container;
+  }, [useSafeArea, animated, reanimated]);
```
Comment (M-i-k-e-l) [inline src/components/view/index.tsx:101]: I can see `useSafeArea` changing, but probably not the other two

File: typings/globalTypes.d.ts [hunk 3]
```diff
@@ -14,3 +14,7 @@ declare module 'react-native-measureme';
 interface Extendable {
   [key: string]: any;
 }
+
+interface ThemeComponent {
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply typings/globalTypes.d.ts:18]: Removed

File: src/components/view/index.tsx [hunk 4]
```diff
@@ -46,106 +40,90 @@ export interface ViewProps extends Omit<RNViewProps, 'style'>, ContainerModifier
...
-    } else if (props.animated) {
-      this.Container = Animated.createAnimatedComponent(this.Container);
-    }
-
-    this.state = {
-      ready: !props.renderDelay
-    };
-  }
+function View(props: ViewProps, ref: any) {
+  const themeProps = useThemeProps(props, 'View');
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/view/index.tsx:60]: I think that by default the component name is used as display name

File: src/components/view/index.tsx (hunk 2 above)
Comment (ethanshar) [reply src/components/view/index.tsx:101]: I agree, And if they won't change there's no harm having them here (:

File: src/components/view/index.tsx [hunk 5]
```diff
@@ -46,106 +40,90 @@ export interface ViewProps extends Omit<RNViewProps, 'style'>, ContainerModifier
...
-  }
+  }, []);
 
-  // TODO: do we need this?
-  setNativeProps(nativeProps: any) {
-    //@ts-ignore
-    this._root.setNativeProps(nativeProps); // eslint-disable-line
-  }
+  const ViewContainer = useMemo(() => {
+    const container = useSafeArea && Constants.isIOS ? SafeAreaView : RNView;
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/view/index.tsx:92]: I need the container at the beginning cause createAnimatedComponent wraps it

File: src/components/view/index.tsx (hunk 1 above)
Comment (ethanshar) [reply src/components/view/index.tsx]: Done

File: src/components/view/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/view/index.tsx:92]: Oops 🤦‍♂️

File: src/components/view/index.tsx (hunk 2 above)
Comment (M-i-k-e-l) [reply src/components/view/index.tsx:101]: It might have some performance implications, but they are probably minor

**Review by M-i-k-e-l — APPROVED** [review]

**Pull request merged**